components/Authentication/login.jsx

```diff
@@ -60,36 +59,55 @@ const ShowText = styled.Text`
   padding-right: 15px;
 `;
 
+const ViewFiller = styled.View`
+  height: 15px;
+`;
+
 const Login = ({ navigation }) => {
   const [email, setEmail] = React.useState('');
   const [password, setPassword] = React.useState('');
   const [hidePassword, setHidePassword] = React.useState(true);
 
   function loginPress() {
     firebase.auth().signInWithEmailAndPassword(email, password)
+      .then(() => {
+        navigation.navigate('Root');
+      })
       .catch((error) => {
-        const errorCode = error.code;
-        const errorMsg = error.message;
-        switch (errorCode) {
-          case 'auth/wrong-password':
-            alert('Incorrect password.');
-            break;
-
-          default:
-            alert(`Error: ${errorMsg}`);
-        }
+        const errorMessage = error.message;
+        // TODO: this alert looks odd for some reason,
+        // can't remember what an invalid password alert looks like
+        Alert.alert('Error:', errorMessage);
       });
   }
 
   function signupPress() {
     navigation.navigate('Signup');
   }
 
-  firebase.auth().onAuthStateChanged(() => {
-    if (firebase.auth().currentUser) {
-      navigation.navigate('Root');
```

Comment:
Do we need this to auto-login if the user has logged in before?

Reply:
no. that's done in app.js see line 136

Reply:
Does it still work? It doesn't do it for me anymore when I run it locally

Reply:
hmm. Well this is just a listener for auth state change, so if they log in or log out it triggers. So it shouldn't be what is automatically logging you in

Reply:
hm does it still work for you? Maybe it takes a second for Firebase to realize you're logged in so this was catching it previously?

Reply:
Not working for me either, although I'm not sure if it should be remembering you between simulator runs since it is a new package on expo. In theory App.js should be doing that part, after initializing firebase it checks for a user. By default firebase keeps users logged in so that should not be null if they are. But idk I'll look into it more before merging

Reply:
Okay, I think everything else in the PR is ready to merge so once we figure that out you can just re-request my review and I'll approve it